L1Trigger/L1CaloTrigger/plugins/L1CaloJetProducer.cc

```diff
@@ -52,1457 +51,1379 @@ and clusters them within fixed number of trigger towers
-    produces< BXVector<l1t::Tau> >("CaloTaus");
-
+public:
+  explicit L1CaloJetProducer(const edm::ParameterSet &);
+
+private:
+  virtual void produce(edm::Event &, const edm::EventSetup &);
+  //bool cluster_passes_base_cuts(float &cluster_pt, float &cluster_eta, float &iso, float &e2x5, float &e5x5) const;
+  int ecalXtal_diPhi(int &iPhi_1, int &iPhi_2) const;
+  int tower_diPhi(int &iPhi_1, int &iPhi_2) const;
+  int tower_diEta(int &iEta_1, int &iEta_2) const;
+  float get_deltaR(reco::Candidate::PolarLorentzVector &p4_1, reco::Candidate::PolarLorentzVector &p4_2) const;
+  float get_hcal_calibration(float &jet_pt, float &ecal_pt, float &ecal_L1EG_jet_pt, float &jet_eta) const;
+  float get_tau_pt_calibration(float &tau_pt, float &ecal_pt, float &l1EG_pt, float &n_L1EGs, float &tau_eta) const;
+  int loose_iso_tau_wp(float &tau_pt, float &tau_iso_et, float &tau_eta) const;
+
+  double HcalTpEtMin;
+  double EcalTpEtMin;
+  double HGCalHadTpEtMin;
+  double HGCalEmTpEtMin;
+  double HFTpEtMin;
+  double EtMinForSeedHit;
+  double EtMinForCollection;
+  double EtMinForTauCollection;
+
+  // For fetching jet calibrations
+  std::vector<double> jetPtBins;
+  std::vector<double> emFractionBinsBarrel;
+  std::vector<double> absEtaBinsBarrel;
+  std::vector<double> jetCalibrationsBarrel;
+  std::vector<double> emFractionBinsHGCal;
+  std::vector<double> absEtaBinsHGCal;
+  std::vector<double> jetCalibrationsHGCal;
+  std::vector<double> emFractionBinsHF;
+  std::vector<double> absEtaBinsHF;
+  std::vector<double> jetCalibrationsHF;
+
+  // For fetching tau calibrations
+  std::vector<double> tauPtBins;
+  std::vector<double> tauAbsEtaBinsBarrel;
+  std::vector<double> tauCalibrationsBarrel;
+  std::vector<edm::ParameterSet> tauL1egInfoBarrel;
+  std::vector<double> tauAbsEtaBinsHGCal;
+  std::vector<double> tauCalibrationsHGCal;
+  std::vector<edm::ParameterSet> tauL1egInfoHGCal;
+
+  // For storing jet calibrations
+  std::vector<std::vector<std::vector<double>>> calibrationsBarrel;
+  std::vector<std::vector<std::vector<double>>> calibrationsHGCal;
+  std::vector<std::vector<std::vector<double>>> calibrationsHF;
+
+  // For storing tau calibration info
+  std::map<double, std::vector<double>> tauL1egInfoMapBarrel;
+  std::map<double, std::vector<double>> tauL1egInfoMapHGCal;
+  std::vector<double> tauL1egValuesBarrel;  // To preserve ordering
+  std::vector<double> tauL1egValuesHGCal;   // To preserve ordering
+  std::vector<std::vector<std::vector<std::vector<double>>>> tauPtCalibrationsBarrel;
+  std::vector<std::vector<std::vector<std::vector<double>>>> tauPtCalibrationsHGCal;
+
+  bool debug;
+  edm::EDGetTokenT<l1tp2::CaloTowerCollection> l1TowerToken_;
+  edm::Handle<l1tp2::CaloTowerCollection> l1CaloTowerHandle;
+
+  // TF1s defining tau isolation thresholds
+  TF1 isoTauBarrel = TF1("isoTauBarrelFunction", "([0] + [1]*TMath::Exp(-[2]*x))");
+  TF1 isoTauHGCal = TF1("isoTauHGCalFunction", "([0] + [1]*TMath::Exp(-[2]*x))");
```

Comment:
if the functional form is known at compile time, please use a C/C++ function instead of a compile-time `TF1`